debugging/coreclr/dockerDebugConfigurationProvider.ts

```diff
@@ -0,0 +1,250 @@
+ * Copyright (C) Microsoft Corporation. All rights reserved.
+ *--------------------------------------------------------*/
+
+import * as path from 'path';
+import * as vscode from 'vscode';
+import { CancellationToken, DebugConfiguration, DebugConfigurationProvider, ProviderResult, WorkspaceFolder } from 'vscode';
+import { PlatformOS } from '../../utils/platform';
+import { DebugSessionManager } from './debugSessionManager';
+import { DockerManager, LaunchResult } from './dockerManager';
+import { FileSystemProvider } from './fsProvider';
+import { NetCoreProjectProvider } from './netCoreProjectProvider';
+import { OSProvider } from './osProvider';
+import { Prerequisite } from './prereqManager';
+
+interface DockerDebugBuildOptions {
+    context?: string;
+    dockerfile?: string;
+    tag?: string;
+    target?: string;
+}
+
+interface DockerDebugRunOptions {
+    containerName?: string;
+    os?: PlatformOS;
+}
+
+interface DebugConfigurationBrowserBaseOptions {
+    enabled?: boolean;
+    command?: string;
+    args?: string;
+}
+
+interface DebugConfigurationBrowserOptions extends DebugConfigurationBrowserBaseOptions {
+    windows?: DebugConfigurationBrowserBaseOptions;
+    osx?: DebugConfigurationBrowserBaseOptions;
+    linux?: DebugConfigurationBrowserBaseOptions;
+}
+
+interface DockerDebugConfiguration extends DebugConfiguration {
+    appFolder?: string;
+    appOutput?: string;
+    appProject?: string;
+    dockerBuild?: DockerDebugBuildOptions;
+    dockerRun?: DockerDebugRunOptions;
+}
+
+export class DockerDebugConfigurationProvider implements DebugConfigurationProvider {
+    constructor(
+        private readonly debugSessionManager: DebugSessionManager,
+        private readonly dockerManager: DockerManager,
+        private readonly fsProvider: FileSystemProvider,
+        private readonly osProvider: OSProvider,
+        private readonly netCoreProjectProvider: NetCoreProjectProvider,
+        private readonly prerequisite: Prerequisite) {
+    }
+
+    public provideDebugConfigurations(folder: WorkspaceFolder | undefined, token?: CancellationToken): ProviderResult<DebugConfiguration[]> {
+        return [
+            {
+                name: 'Docker: Launch .NET Core ASP.NET (Preview)',
+                type: 'docker-coreclr',
+                request: 'launch',
+                preLaunchTask: 'build',
+                dockerBuild: {
+                },
+                dockerRun: {
+                }
+            }
+        ];
+    }
+
+    public resolveDebugConfiguration(folder: WorkspaceFolder | undefined, debugConfiguration: DockerDebugConfiguration, token?: CancellationToken): ProviderResult<DebugConfiguration | undefined> {
+        return this.resolveDockerDebugConfiguration(folder, debugConfiguration);
+    }
+
+    private static resolveFolderPath(folderPath: string, folder: WorkspaceFolder): string {
+        return folderPath.replace(/\$\{workspaceFolder\}/g, folder.uri.fsPath);
```

Comment:
should this be /gi for case-insensitive?

Reply:
Reasonable; updated.